File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        total = 0
        max_time = 0
        n = len(colors)
        
        for i in range(n):
            if i > 0 and colors[i] != colors[i - 1]:
                max_time = 0
            total += min(max_time, neededTime[i])
            max_time = max(max_time, neededTime[i])
        
        return total

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cs
public class Solution
{
    public int MinCost(string colors, int[] neededTime)
    {
        int total = 0;
        int maxTime = neededTime[0];
        
        for (int i = 1; i < colors.Length; i++)
        {
            if (colors[i] != colors[i - 1])
            {
                maxTime = 0;
            }
            total += (maxTime < neededTime[i] ? maxTime : neededTime[i]);
            maxTime = (maxTime > neededTime[i] ? maxTime : neededTime[i]);
        }
        
        return total;
    }
}